### PR TITLE
Configure BT UUID Via .INI Entry

### PR DESCRIPTION
**Pull request by JackLivio**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Pair mobile device over bluetooth, and connect an SDL app using the bluetooth transport.

### Summary
Add a new .ini configuration line to allow the SDL bluetooth UUID to be set at runtime instead of compiled with Core.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/include/transport_manager/transport_manager_settings.h [hunk 1]
```diff
@@ -79,6 +79,8 @@ class TransportManagerSettings : public TransportManagerMMESettings {
    * @brief Returns maximum retry attempts for cloud app connections
    */
   virtual uint16_t cloud_app_max_retry_attempts() const = 0;
+
+  virtual const uint8_t* bluetooth_uuid() const = 0;
```
Comment (jacobkeeler) [inline src/components/include/transport_manager/transport_manager_settings.h]: Build fails with tests on

File: src/components/include/application_manager/application_manager_settings.h [hunk 2]
```diff
@@ -88,6 +88,7 @@ class ApplicationManagerSettings : public RequestControlerSettings,
   virtual uint32_t rpc_pass_through_timeout() const = 0;
   virtual uint32_t cloud_app_retry_timeout() const = 0;
   virtual uint16_t cloud_app_max_retry_attempts() const = 0;
+  virtual const uint8_t* bluetooth_uuid() const = 0;
```
Comment (jacobkeeler) [inline src/components/include/application_manager/application_manager_settings.h]: Is this needed in `ApplicationManagerSettings`?

File: src/components/config_profile/src/profile.cc [hunk 3]
```diff
@@ -1821,6 +1826,9 @@ void Profile::UpdateValues() {
                     kCloudAppMaxRetryAttemptsKey,
                     kCloudAppTransportSection);
 
+  bluetooth_uuid_ =
+      ReadUint8Container(kTransportManagerSection, kBluetoothUUIDKey, NULL);
```
Comment (jacobkeeler) [inline src/components/config_profile/src/profile.cc]: Should probably include validation that the list is exactly 16 bytes long

File: src/components/include/transport_manager/transport_manager_settings.h (hunk 1 above)
Comment (jacobkeeler) [reply src/components/include/transport_manager/transport_manager_settings.h]: After fixing build failures, transport_manager_test segfaults

File: src/appMain/smartDeviceLink.ini [hunk 4]
```diff
@@ -202,6 +202,9 @@ TCPAdapterPort = 12345
 ; If the name is omitted, Core will listen on all network interfaces by binding to INADDR_ANY.
 TCPAdapterNetworkInterface =
 
+; 128 bit uuid for bluetooth device discovery. Please format as 16 seperate bytes.
+BluetoothUUID = 0x93, 0x6D, 0xA0, 0x1F, 0x9A, 0xBD, 0x4D, 0x9D, 0x80, 0xC7, 0x02, 0xAF, 0x85, 0xC8, 0x22, 0xA8
```
Comment (jacobkeeler) [inline src/appMain/smartDeviceLink.ini]: Commenting this out results in a segfault

File: src/components/transport_manager/src/bluetooth/bluetooth_device_scanner.cc [hunk 5]
```diff
@@ -137,6 +137,26 @@ BluetoothDeviceScanner::BluetoothDeviceScanner(
...
+    , shutdown_requested_(false)
+    , ready_(true)
+    , device_scan_requested_(false)
+    , device_scan_requested_lock_()
+    , device_scan_requested_cv_()
+    , auto_repeat_search_(auto_repeat_search)
+    , auto_repeat_pause_sec_(auto_repeat_pause_sec) {
+  sdp_uuid128_create(&smart_device_link_service_uuid_,
+                     smart_device_link_service_uuid_data);
+  thread_ = threads::CreateThread("BT Device Scaner",
```
Comment (jacobkeeler) [inline src/components/transport_manager/src/bluetooth/bluetooth_device_scanner.cc]: might as well fix this typo

File: src/appMain/smartDeviceLink.ini (hunk 4 above)
Comment (jacobkeeler) [reply src/appMain/smartDeviceLink.ini]: Should include this as a default value in `profile.cc` and comment this line out by default.

**Comment by jacobkeeler** [comment]
Also, fix style

**Comment by JackLivio** [comment]
@jacobkeeler All comments have been addressed

**Pull request merged**